### ISSUE TITLE
Auto-added Google Profile overwriting contact name

### ISSUE THREAD

**Issue by Rotaryracer** [zanonmark/Google-4-TbSync#8]
Hi Mark - THANK YOU for picking up the charge and integrating Google Contacts into Thunderbird.  I've been using this for a few weeks now in read/write mode (after backing up my contacts) and it's been working great. 

One thing I did notice yesterday was that my personal contact info in my phone was overwritten by a Google Profile that was auto-added to my Google Contact.  I've tried overwriting the contact name in Thunderbird, but that didn't work.  I overwrote the contact name in Google Contacts and that did work to change there, but when I did my next sync, it overwrote the TBird contact.  I've tried to disconnect the Google profile from the Google Contact, but they seem to be stuck together with no easy way to disconnect.  I even removed the email address that is said caused the auto-add, and it just used my backup recovery email (which I still need in the contact) to auto-add.  

Anyway - not sure if there's a way to ensure Google-4-TbSync only adds contact first/last name and uses as display name, versus pulling the Google Profile name?  Not a big deal for my personal contact, but if this is a new Google thing and they start auto-adding profile data (and profiles don't match the names I have), it could get interesting.

Thanks again...

Jason

**Comment by zanonmark** [comment]
Hi Jason, and thanks for Your report.

> Anyway - not sure if there's a way to ensure Google-4-TbSync only adds contact first/last name and uses as display name, versus pulling the Google Profile name? Not a big deal for my personal contact, but if this is a new Google thing and they start auto-adding profile data (and profiles don't match the names I have), it could get interesting.

The list of properties I download from the server are in `AddressBookSynchronizer.js`, in the `fillLocalContactWithServerContactInformation()` method.
Particularly, for names, Thunderbird tries to retrieve: `FirstName`, `LastName` and `DisplayName`. Maybe is this last one causing troubles?

Please let me know so we can work it out.

Thanks,
MZ

**Comment by Rotaryracer** [comment]
Hi Mark - thanks for the reply.  For my Google profile, I use the two words of my company name (we'll call it Biz Name) as my first and last name.  In Google Contacts, I had my personal contact details listed correctly with my actual name, but because I added my work email address, Google "helpfully" glued my profile to it.  When it did so (which appears to have been a recent thing), it changed my personal record in TBird to Biz Name, versus my actual name.  

I played with it a bit more this AM and sort of have a workaround.  Basically, if I delete and then re-add all the same info in Google Contacts, it populates and adds the profile, which then means TBird picks it up as Biz Name.  If I leave all of my business details out of the Google Contacts record, it doesn't match me up with the Google profile, and keeps my name accurate.

One other thing I did note is that when I created a new Google Contacts record for me (without biz info, but with all personal details - home address, etc), it didn't seem to sync all the fields.  I can see my home address in Google, but it doesn't carry over to TBird.  For grins, I put a test address in TBird to see if it would overwrite what's in Google, but it did not do so.  It could be an issue with my specific record, since I've tried to delete and re-add both locally and on Google, but thought I would mention it.

Thanks!

Jason

**Comment by zanonmark** [comment]
> For my Google profile, I use the two words of my company name (we'll call it Biz Name) as my first and last name. In Google Contacts, I had my personal contact details listed correctly with my actual name, but because I added my work email address, Google "helpfully" glued my profile to it. When it did so (which appears to have been a recent thing), it changed my personal record in TBird to Biz Name, versus my actual name.

ok, so what You get in Thunderbird (by double-clicking the record for example) is:
First name: Jason
Last name: YourLastName
Display name: Biz Name
is it right?

or do You get:
First name: Biz
Last name: Name
Display name: Biz Name
?

> I played with it a bit more this AM and sort of have a workaround. Basically, if I delete and then re-add all the same info in Google Contacts, it populates and adds the profile, which then means TBird picks it up as Biz Name. If I leave all of my business details out of the Google Contacts record, it doesn't match me up with the Google profile, and keeps my name accurate.

Yes, I suppose Google is messing up the data by merging two different "worlds".
I'll probably ask You to kindly debug a little bit more once I know what You read in Thunderbird (see question above).

> One other thing I did note is that when I created a new Google Contacts record for me (without biz info, but with all personal details - home address, etc), it didn't seem to sync all the fields. I can see my home address in Google, but it doesn't carry over to TBird. For grins, I put a test address in TBird to see if it would overwrite what's in Google, but it did not do so. It could be an issue with my specific record, since I've tried to delete and re-add both locally and on Google, but thought I would mention it.

Well, first of all, thanks for reporting.
If You edit the same contact remotely and locally, there shouldn't be any issue because the remote version always wins. Obviously, a bug can't be excluded :)

However, I would postpone this issue a little, because it might be related to the first one You reported - so, let's try to fix it first, then this one (which, incidentally, if related to any synchronization problem, could potentially be fixed automatically when rewriting the add-on in a few weeks after TbSync is ported to the WebExtension API, because this part of the code is actually the one that will be rewritten nearly from scratch).

Thanks,
MZ

**Comment by zanonmark** [comment]
Hi Jason,

could You please confirm what You get:

First name: Jason
Last name: YourLastName
Display name: Biz Name

or

First name: Biz
Last name: Name
Display name: Biz Name
?

Thank You very much for Your report,
MZ

**Comment by Rotaryracer** [comment]
Sorry for the delay...after Google appended the business profile to my Google contact, it was displaying the second option:

First name: Biz
Last name: Name
Display name: Biz Name

Regardless of changing and ensuring it was my correct name in Google, it kept updating to "Biz Name" in T-Bird.

Thanks!

Jason

**Comment by zanonmark** [comment]
> First name: Biz
> Last name: Name
> Display name: Biz Name

Thanks

> Regardless of changing and ensuring it was my correct name in Google, it kept updating to "Biz Name" in T-Bird.

First test: could You please add some new information too, e.g.: modify the names AND add another phone number, or website, or whatever? Then redo the synchronization, and see what happens...

Thanks,
MZ

**Comment by zanonmark** [comment]
Could You please test with the latest version in the current development tree?

Thanks,
MZ

**Comment by Rotaryracer** [comment]
I temporarily uninstalled as I tested TBSync on a hosted CardDAV server, but am finding that has other issues. :)  I will reinstall the newest version and update on results. 

Thanks!

Jason

**Comment by Rotaryracer** [comment]
I had wiped all my contacts from Google, so just exported and imported.  I installed from the T-Bird Add-On page and (so far), it seems to be synchronizing without issue.  I'm not sure if that will change when/if Google tries to append my business name to my contact, but fingers crossed!  I'm also not sure if the Add-On version is the same as what's in the development tree now, but if not (and if I see that Google glitch return), I'll try to update to the newest.

Thanks again!

Jason
